factorize rootfs filelist and image generation

etienne-lms opened the pull request:
factorize also all rootfs generation between fvp/juno/qemus/mediatek

jbech-linaro commented:
This is a good idea, which devices have you tested this on so far?

etienne-lms commented on mediatek.mk:
```diff
@@ -112,36 +112,13 @@ helloworld-clean: helloworld-clean-common
-	@echo "dir /lib/aarch64-linux-gnu 755 0 0" >> $(GEN_ROOTFS_FILELIST)
-	@echo "file /lib/aarch64-linux-gnu/libteec.so.1.0 $(OPTEE_CLIENT_EXPORT)/lib/libteec.so.1.0 755 0 0" >> $(GEN_ROOTFS_FILELIST)
-	@echo "slink /lib/aarch64-linux-gnu/libteec.so.1 libteec.so.1.0 755 0 0" >> $(GEN_ROOTFS_FILELIST)
-	@echo "slink /lib/aarch64-linux-gnu/libteec.so libteec.so.1 755 0 0" >> $(GEN_ROOTFS_FILELIST)
```
sorry: i missed this part: specific library path.
i will fix.

etienne-lms replied:
Actually this specific library path is inherited from pre commit 69a8a3-"gcc5: Update to use gcc5".
mediatek should also use the standard /lib" path. I will not "fix" the PR for specific path, contrary to what I stated above.

jbech-linaro replied:
Yes, that is true pre gcc5, there we differences between 32 and 64 bit builds. But they should be the same since then. Since the platform hasn't been verified, I'm OK, with postpone this to later point in time if and when we put efforts into getting the MTK board up and running again.

etienne-lms commented:
i only tested this over qemu and qemu_v8 (hence an issue in mediatek lib path)

i made this change to ease SMAF test setup integration (several libs, drv, tools to embedded).
sqlfs integration also shows this can be slightly useful.

jbech-linaro commented:
MTK device is currently not that important, since that hasn't been verified since OP-TEE 2.0.0 (it's mentioned in the the release notes). The FS generation is straight forward, but it's quite easy to make mistakes when doing bulk changes, so it would probably be good to have a couple of "Tested-by" tags on this particular patch before merging. All in all, I like this.

jforissier commented:
`Tested-by: Jerome Forissier <jerome.forissier@linaro.org> (HiKey)`

etienne-lms commented:
`Tested-by: Etienne Carriere <etienne.carriere@linaro.org> (qemu_v7/v8)`

jbech-linaro commented:
I haven't tried, but I think this would break the RPi3, stuff since parts of that build isn't on github.com/OP-TEE. Since we have some demos at conferences in the pipe related to RPi3, I'd like to ask wait merge this until I've made pull requests for RPi3 (we're talking about build and optee_os mainly), i.e. just so we know that the RPi3 setup is working (we have no stable tag to lean back to).

etienne-lms commented:
ok, no hurry.

jforissier commented:
@jbech-linaro fine with me

jbech-linaro commented:
Great, I'll try to make PR's (where I mentioned this PR) as soon as possible, so this shouldn't be pending for too long (I hope).

etienne-lms commented:
rebased.
updated with Raspeberry Pi 3.
This change also merges proposed changes from #77.

joakim: could you test this on the RPI3 ?
I will check on qemu, qemu_v8 and our ST b2120/b2260 boards.
Jerome already tested this on HiKey. No new change so it should be still be ok.

etienne-lms commented on common.mk:
```diff
@@ -249,3 +249,73 @@ HELLOWORLD_CLEAN_COMMON_FLAGS ?= TA_DEV_KIT_DIR=$(OPTEE_OS_TA_DEV_KIT_DIR)
+	rm -f $(GEN_ROOTFS_PATH)/filelist-all.txt
+	rm -f $(GEN_ROOTFS_PATH)/filelist-tmp.txt
+
+#.PHONY: filelist-tee-common
```
hum... i should restore this `.PHONY: filelist-tee-common`

etienne-lms commented on rpi3.mk:
```diff
@@ -190,34 +190,11 @@ helloworld-clean: helloworld-clean-common
+
+filelist-tee: xtest helloworld
+filelist-tee: filelist-tee-common
 	@echo "dir /usr/bin 755 0 0" >> $(GEN_ROOTFS_FILELIST)
```
is this directory creation really required here ?

jbech-linaro commented:
@etienne-lms , yes, I can try RPi3. Hopefully I find a slot during the day. Otherwise it has to wait to next week when I'm back from my trip.

etienne-lms commented:
@jbech-linaro: fine. anyhow, i'm sure i'll need to change something or another and ask you to check back. so don't hurry (and be happy).

jbech-linaro commented:
Without digging into it, something didn't work.
```
make[1]: Leaving directory '/mnt/sshd/devel/optee_projects/rpi3/linux'
cat /mnt/sshd/devel/optee_projects/rpi3/build/../gen_rootfs/filelist-final.txt > /mnt/sshd/devel/optee_projects/rpi3/build/../gen_rootfs/filelist.tmp
cat: /mnt/sshd/devel/optee_projects/rpi3/build/../gen_rootfs/filelist-final.txt: No such file or directory
make: *** [common.mk:257: update_rootfs-common] Error 1
```

etienne-lms commented:
rebased on #77 merge.
cleanup useless dependancies.
propose log level sync.
fix qemu_v8 and rpi3 (hopefuly) dependaencies.

tested on quem_v8 & qemu

etienne-lms commented on common.mk:
```diff
@@ -217,7 +218,7 @@ optee-client-clean-common:
 	CROSS_COMPILE_TA=$(CROSS_COMPILE_S_USER) \
 	TA_DEV_KIT_DIR=$(OPTEE_OS_TA_DEV_KIT_DIR) \
-	CFG_DEV_PATH=$(ROOT) \
+	OPTEE_CLIENT_EXPORT=$(OPTEE_CLIENT_EXPORT)
```
arg! missing `\` at end of line.

jforissier commented:
`Tested-by: Jerome Forissier <jerome.forissier@linaro.org> (HiKey)`

jbech-linaro commented:
I haven't my RPi3 here right now, so I cannot test it on the device, but with the current patch it doesn't complain any longer, when I'm building.
`Acked-by: Joakim Bech <joakim.bech@linaro.org>`

etienne-lms commented:
@jbech-linaro, we can wait for you to test this on RPi3 before merging.

jbech-linaro commented:
Ok, I'll test RPi3 also. Since I'm at the office, I've tried this on the MTK8173 and that seems to be working fine. So for now and for that device.
`Tested-by: Joakim Bech <joakim.bech@linaro.org> (MTK8173)`

jbech-linaro commented:
RPi3 also works fine, so with current commit:
`Reviewed-by: Joakim Bech <joakim.bech@linaro.org>`
`Tested-by: Joakim Bech <joakim.bech@linaro.org> (RPi3, MTK8173)`
